actionview/lib/action_view/railtie.rb

```diff
@@ -48,6 +48,14 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "action_view.eager_load_templates" do |app|
+      ActiveSupport.on_load(:action_controller) do
+        if Rails.env.production? && app.config.action_view.eager_load_templates
```

Comment:
We use custom `staging` environment which mimics production but not using same as production because _some_ things need to be different (like secret keys for 3rd party apis), but it's running on production-like hardware and all possible config should be same as production, to reduce chance failure after deploy. Many production applications are using this setup. So hardcoding a fixed environment will not work.

If one of your config depends on another config, you should simply check for it at runtime and raise  an exception if you detect incompatible configuration.